build.gradle

```diff
@@ -160,56 +250,60 @@ task zip(type: Zip) {
     }
 }
 
-task installPlugin() {
-    if (!new File("${rootDir}/dist").exists()) {
-        return
-    }
-    String[] libJars = java.util.Optional.ofNullable(new File("${rootDir}/dist/lib").list()).orElseGet(() -> new String[0])
-    getAllprojects().forEach(subProject -> {
-        var file = new File("${subProject.projectDir}/gradle.properties")
-        if (!file.exists()) {
+tasks.register('installPlugin') {
+    doLast {
+        if (!new File("${rootDir}/dist").exists()) {
             return
         }
```

Comment:
I don't quite grasp the alterations made on `dist` and `installPlugin` tasks. There have been several changes in grammar, such as why "doLast" was added in this case?

Reply:
I noticed that some of your tasks (like `printProjects`) are executed regardless of what I put on the command line.

I am not a Gradle expert, so I followed the [Task Configuration Avoidance](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) guide.

From my experiments:

```groovy
task installPlugin() {
    println "Hello world!"
}
```

is executed **always**, whereas:

```groovy
tasks.register('installPlugin') {
    doLast {
        println "Hello world!"
    }
}
```
is executed only if the task is on the command line.

Reply:
It seems `tasks.register('installPlugin')` equals to `task installPlugin()` and is a newer writting.

I'm afraid that `doLast` will change the original execution sequence. I have to test `dist` and `installPlugins` task to make sure they keep the same behavior with the original one.

Reply:
@Pil0tXia,

I compared your release 1.10.0 with the one generated by this script and they only differ regarding their logging dependencies (as expected).

Reply:
Good job!

"Task configuration avoidance API" is an updated writting and has backward compatibility. IDEA is also warning the outdated writting, so the new writting should be adopted.

![image](https://github.com/apache/eventmesh/assets/41445332/e637068f-4dcc-4beb-a483-b040af9fa759)

---

`doLast()` is also a newer writting, and should be adopted. However, the `printProjects` task is meant to print all modules before each task executes, so I think it doesn't need to apply `doLast()`.

So maybe like this:

![image](https://github.com/apache/eventmesh/assets/41445332/2a54c1ab-f7f4-4f9e-875b-8e1296e61612)

[What is doLast() for? - Help/Discuss - Gradle Forums](https://discuss.gradle.org/t/what-is-dolast-for/27731/3)